src/components/phylogeny/Phylotree.vue

```diff
@@ -324,169 +359,213 @@ export default {
+
+            // Add a custom menu item to allow us to rename this node.
+            // console.log("node", node);
+            addCustomMenu(
+              node,
+              (node) => "Rename this node",
+              () => {
+                const node = data;
+                const existingName = node.name || "(none)";
+                const newName = window.prompt(
+                  `Rename node named '${existingName}' to:`
                 );
-                textLabel.classed("selected-internal-label", true);
-              }
-            } else if (!textLabel.empty()) textLabel.remove();
-          }
+                if (newName === null) {
+                  // This means the user clicked "Cancel", so don't do anything.
+                } else if (!newName || newName === "undefined") {
+                  // Apparently IE7 and IE8 will return the string 'undefined' if the user doesn't
+                  // enter anything.
+                  //
+                  // Remove the current label.
+                  node.name = "";
+                } else {
+                  // Set the new label.
+                  node.name = newName;
+                }
 
-          // Clear any existing menu items.
-          node.menu_items = [];
-
-          // Add a custom menu item to allow us to rename this node.
-          console.log("node", node);
-          addCustomMenu(
-            node,
-            (node) => "Rename this node",
-            () => {
-              const node = data;
-              const existingName = node.name || "(none)";
-              const newName = window.prompt(
-                `Rename node named '${existingName}' to:`
+                // Export the entire phylogeny as a Newick string, and store that
+                // in the phylogeny object.
+                const updatedNewickString = this.tree.getNewick();
+                console.log("updatedNewickString", updatedNewickString);
+                this.$store.commit("setPhylogenyProps", {
+                  phylogeny: this.phylogeny,
+                  newick: updatedNewickString,
+                });
+              },
+              (node) => true // We can replace this with a condition that indicates whether this node should be displayed.
+            );
+
+            // If the internal label has the same IRI as the currently selected
+            // phyloreference's reasoned node, further mark or label it as the resolved node.
+            //
+            // Note that this node might NOT be labeled, in which case we need to
+            // label it now!
+            if (
+              phyloref !== undefined &&
+              has(data, "@id") &&
+              this.$store.getters
+                .getResolvedNodesForPhylogeny(this.phylogeny, phyloref)
+                .includes(data["@id"])
+            ) {
+              // We found another pinning node!
+              if(!(data["@id"] in phylorefsByNodeId)) phylorefsByNodeId[data["@id"]] = new Set();
```

Comment:
Indent the statement under the conditional? Perhaps having it on the same line has been part of our coding style all along but it did surprise me here and took looking thrice to understand while scrolling through.

Reply:
Done in d476dbd525cfbf3b6b397e252540afeed33d2dfd.

I haven't used our code linter in a while >.< Once we're done with the Klados 1.0 work, I should run it to see what can be cleaned up automatically/look into integrating that into PRs automatically.